kaybee/cmd/export.go

```diff
@@ -51,11 +51,11 @@ func doExport(args []string) {
 	// }
 
 	t := tasks.NewExportTask().
-		WithExportScripts(configuration.ExportScripts()).
+		WithExportScripts(configuration.GetExportScripts()).
```

Comment:
This and the following are against (golang) best practices. Less is more, drop the "get" from getters:
https://golang.org/doc/effective_go.html#Getters

Reply:
Yes, I'm aware that calling it `ExportScripts()` would be better but this was a workaround because of two issues:

- If a struct with attribute `A` cannot implement an interface with a function `A()`. The compiler would assume that it's a duplicate definition of `A` (which is weird and isn't documented anywhere).

- I wanted to change the attribute to `a` and keep the function as `A()`. However this introduces another issue with the `yaml` unmarshal (which can only export and import exported attributes).

Reply:
Right, but that problem was solved in my previous architecture by decoupling syntax-oriented structs (ConfV1, etc) from internal normalized configuration object (which can have lowercase attribute a and upper-case getter A()).

I guess your solution aims to achieve more than mine can give: allow evolution of the information content in the configuration file whereas mine only caters for shallow syntactic evolution. As I wrote in another comment, your solution gives more, but costs more (requires version-specific behaviour down to the task implementations, which makes components with very different responsibilities dependent on each other); I am not sure we really need that.

Especially at this stage of the project, I would keep things simple and focus on the high-return critical features that are still missing (in no particular order):
* https://github.com/SAP/project-kb/issues/22
* https://github.com/SAP/project-kb/issues/20
https://github.com/SAP/project-kb/issues/6